brave-core/src/main/java/com/github/kristofa/brave/LocalTracer.java

```diff
@@ -83,7 +83,10 @@ public SpanId startNewSpan(String component, String operation) {
     }
 
     private SpanId getNewSpanId() {
-        Span parentSpan = spanAndEndpoint().state().getCurrentServerSpan().getSpan();
+        Span parentSpan = spanAndEndpoint().state().getCurrentLocalSpan();
```

Comment:
Again, users currently don't need to close spans in any of the tracers. For example, they can report them manually, and there's no instructions against that. The key here is to say on the javadocs of the new type that you *must* finish spans, or at least clear the span state. It must also clarify that clearing the span state expects to pop the previous span. For example, right now, it is harmless to clear the span state multiple times. In the new impl, clearing has pop semantics.

Bottom line, I think it is a very clever way to accomplish the goal in Brave 3, and it is good to have a workaround, but we have to be very careful with it as the state store api in Brave 3 wasn't made for stack semantics. By guarding on this, we say "proceed at your own risk" which certainly won't break others.